scripts/test/SANS/gui_logic/run_tab_presenter_test.py

```diff
@@ -267,6 +268,22 @@ def test_fails_silently_when_batch_file_does_not_exist(self):
         self._remove_files(user_file_path=user_file_path)
         self.os_patcher.start()
 
+    def test_batch_file_dir_not_added_to_config(self):
+        self.os_patcher.stop()
+        presenter = RunTabPresenter(SANSFacility.ISIS)
+        user_file_path = create_user_file(sample_user_file)
+        view, settings_diagnostic_tab, masking_table = create_mock_view(user_file_path, "A/Path/batch_file.csv")
+        presenter.set_view(view)
+
+        presenter.on_batch_file_load()
+        config_dirs = config["datasearch.directories"]
+        result = "A/Path/" in config_dirs
+
+        self.assertFalse(result, "We do not expect A/Path/ to be added to config, "
+                                 "datasearch.directories is now {}".format(config_dirs))
+
+        self.os_patcher.start()
```

Comment:
I don't think we need to start the patcher at the end as setUp is run before every test anyway. If we did want to restart it I think the line `self.osMock = self.os_patcher.start()` would be better as this sets the mock up again as well.